C++: Add query for missing mode argument in `open`/`openat` calls (#7798)
github/codeql: cpp/ql/lib/change-notes/2022-02-02-detect-creat-in-openat.md

---
category: minorAnalysis
---

> * he `cpp/world-writable-file-creation` query now only detects `open` and `openat` calls with the `O_CREAT` or `O_TMPFILE` flag.

Reviewer: ```suggestion
* The `cpp/world-writable-file-creation` query now only detects `open` and `openat` calls with the `O_CREAT` or `O_TMPFILE` flag.
```

> * The `cpp/world-writable-file-creation` query now only detects `open` and `openat` calls with both the `O_CREAT` and `O_TMPFILE` flag, irrespective of whether the `mode` argument is present.

Reviewer 2: This is getting difficult to understand.  Is the following still accurate?
```suggestion
* The `cpp/world-writable-file-creation` query now only detects `open` and `openat` calls with the `O_CREAT` or `O_TMPFILE` flag.
```

Author: Close enough. I'll change it.
